### ISSUE TITLE
[New rule] dispose-fields

### ISSUE THREAD

**Issue by fzyzcjy** [dart-code-checker/dart-code-metrics#1062]
**Please describe what the rule should do:**

It is common for flutter dev to forget disposing things, so this may be helpful for devs. Maybe we can also scan flutter repo itself using this linter rule :)

Consider the following case:

```dart
class Hello extends StatefulWidget {
  const Hello({Key? key}) : super(key: key);

  @override
  State<Hello> createState() => _HelloState();
}

class _HelloState extends State<Hello> {
  AnimationController? controller;
// OR this
  final controller = AnimationController();
// OR this
  late final controller = AnimationController();
// ... etc
  
  @override
  void initState() {
    super.initState();
    controller = AnimationController();
  }
  
  @override
  void dispose() {
    controller.dispose(); // <---
    super.dispose();
  }
  
  @override
  Widget build(BuildContext context) {
    return Container();
  }
}
```

If we forget to call `dispose`, it will cause memory leak or other bad things.

More formally, it should lint if the following satisfies:

* The type of a field has a `dispose` method (e.g. AnimationController, ui.Image, ... a ton of things)
* The class itself has a `dispose` method (e.g. the State class above)
* The `field.dispose` method is not called in `classinstance.dispose` method

**If your rule is inspired by other please provide link to it:**

**What category of rule is this? (place an "X" next to just one item)**

- [ ] Warns about a potential error (problem)
- [ ] Suggests an alternate way of doing something (suggestion)
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

```dart

```

**Are you willing to submit a pull request to implement this rule?**

**Comment by incendial** [comment]
It's not the same as https://dartcodemetrics.dev/docs/rules/flutter/always-remove-listener?

**Comment by fzyzcjy** [comment]
Looks close but somehow different: this one is for general `dispose`. For example `ui.Image` should be disposed (while it is not a listener at all). But I guess the code will be very similar to that

**Comment by incendial** [comment]
Available in Teams 1.3.0, https://dcm.dev/docs/teams/rules/flutter/dispose-fields/
